doc/guides/debugging/index.rst

```diff
@@ -1,7 +1,7 @@
-.. _debugging:
+.. _flashing_and_debugging:
```

Comment:
Docs changes look good. Just one last thing related to the documentation: I think it would be better to move this directory to `guides/flash-debug` or so.

Reply:
Sure, no problem!